server/endpoint/status/endpoint.go

```diff
@@ -0,0 +1,158 @@
+package status
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	kitendpoint "github.com/go-kit/kit/endpoint"
+	kithttp "github.com/go-kit/kit/transport/http"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Method is the HTTP method this endpoint is register for.
+	Method = "PATCH"
+	// Name identifies the endpoint. It is aligned to the package path.
+	Name = "status/updater"
+	// Path is the HTTP request path this endpoint is registered for.
+	Path = "/status/{app_namespace}/{app_name}/"
+)
+
+type Config struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+}
+
+type Endpoint struct {
+	k8sClient k8sclient.Interface
+	logger    micrologger.Logger
+}
+
+func New(config Config) (*Endpoint, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+
+	e := &Endpoint{
+		logger:    config.Logger,
+		k8sClient: config.K8sClient,
+	}
+
+	return e, nil
+}
+
+func (e Endpoint) Decoder() kithttp.DecodeRequestFunc {
+	return func(ctx context.Context, r *http.Request) (interface{}, error) {
+		var request Request
+
+		defer r.Body.Close()
+		err := json.NewDecoder(r.Body).Decode(&request)
+		if err != nil {
+			return nil, microerror.Maskf(decodeFailedError, "%v", err.Error())
+		}
+
+		namespace, _ := ctx.Value("app_namespace").(string)
+		name, _ := ctx.Value("app_name").(string)
+		request.AppNamespace = namespace
```

Comment:
This is not fancy like `api` repository does, but I would like to keep it as simple.

Reply:
Yes api has middlewares and other stuff we don't need. Good call to keep it simple.